src/plugins/explore/public/components/query_panel/components/editor_stack/resuable_editor.tsx

```diff
@@ -87,6 +87,19 @@ export const ReusableEditor: React.FC<ReusableEditorProps> = ({
 
   const handleEditorDidMount = useCallback(
     (editor: monaco.editor.IStandaloneCodeEditor) => {
+      // Focus on mount
+      editor.focus();
+      setEditorIsFocused(true);
+
+      // set cursor to the end of the document
+      const model = editor.getModel();
+      if (model) {
+        const lastLine = model.getLineCount();
+        const lastColumn = model.getLineMaxColumn(lastLine);
+        editor.setPosition({ lineNumber: lastLine, column: lastColumn });
+        editor.revealPositionInCenter({ lineNumber: lastLine, column: lastColumn });
+      }
```

Comment:
ah i see so its not the actual components thats getting switched but just css getting swithcec. i'm fine with that, but my question is why this logic needed? Why does the editor lose focus when switching?